spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/IcebergPropertiesConverter.java

```diff
@@ -5,19 +5,52 @@
 
 package com.datastrato.gravitino.spark.connector.iceberg;
 
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.IcebergTablePropertiesMetadata;
+import com.datastrato.gravitino.shaded.com.google.common.collect.ImmutableSet;
 import com.datastrato.gravitino.spark.connector.PropertiesConverter;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
 
 /** Transform Iceberg catalog properties between Spark and Gravitino. */
 public class IcebergPropertiesConverter implements PropertiesConverter {
+
+  private static final Set<String> RESERVED_PROPERTIES =
+      ImmutableSet.of(IcebergTablePropertiesMetadata.PROVIDER);
+
   @Override
   public Map<String, String> toGravitinoTableProperties(Map<String, String> properties) {
-    return new HashMap<>(properties);
+    return rebuildCreateProperties(properties);
   }
 
   @Override
   public Map<String, String> toSparkTableProperties(Map<String, String> properties) {
     return new HashMap<>(properties);
   }
+
+  private Map<String, String> rebuildCreateProperties(Map<String, String> createProperties) {
```

Comment:
seems not related to partition?

Reply:
https://github.com/datastrato/gravitino/pull/2709/files#diff-910d75c3ef3beafad853cbf648d14be9845219735600ecf24de76df5455b7ec0L189-L192

properties-related, i removed the limitation above. Otherwise `using parquet` will fail in `testCreateDatasourceFormatPartitionTable`.

so i made a new limitation here.

And `gravitinoTable`'s properties have no `location` by default, so i got it from `SparkTable`'s properties to avoid some `ITs` failures.

Reply:
maybe we can create a new issue, and put it in the same pr.

Reply:
will split a new pr about iceberg propertis. cc @FANNG1